util/dtgen/helper.py

```diff
@@ -196,6 +197,50 @@ def render_value(self, value: dict[Name, object]) -> str:
         return "{\n" + indent_text(text, "  ") + "}"
 
 
+class Extension(ABC):
+    """
+    Base class for extensions.
+    """
+    @abstractmethod
+    def ipname(self) -> str:
+        """
+        Override this function to return the name of the IP to consider.
+        """
+
+    def extend_dt_ip(self, ip_helper: "IpHelper") -> "StructType":
```

Comment:
Quoted type annotation is only required if the type isn't defined at the point when the line is executed i.e. method that returns an object of the class the method belongs to.

```suggestion
    def extend_dt_ip(self, ip_helper: "IpHelper") -> StructType:
```